engines/sci/engine/script_patches.cpp

```diff
@@ -1515,6 +1515,134 @@ static const uint16 gk1Day5MoselyVevePointsPatch[] = {
 	PATCH_END
 };
 
+// When entering the police station (room 230) sGabeEnters sets ego speed
+//  to 4 for the door animation but fails to restore it to the game speed
+//  by calling GKEgo:normalize. This leaves ego at 75% speed until doing
+//  something that does call normalize.
+//
+// We fix this by calling GKEgo:normalize after Gabriel finishes walking
+//  through the door in sGabeEnters:changeState(5). This requires overwriting
+//  the instructions in state 4 which set GKEgo:ignoreActors to 0 but that's
+//  okay because normalize does that.
+//
+// There are two versions of this patch due to two significantly different
+//  versions of this script. The first is in english pc floppy prior to
+//  Sierra's 1.0b patch and the second is in cd and localized floppies. The
+//  script was restructured and the compiler used different sized instructions.
+//
+// Applies to: All PC Floppy and CD versions. TODO: Test Mac, should apply
+// Responsible method: sGabeEnters:changeState
+// Fixes bug #10780
+static const uint16 gk1PoliceEgoSpeedFixV1Signature[] = {
+	0x31, 0x1f,                     // bnt 1f [ state 5 ]
+	SIG_ADDTOOFFSET(+19),
+	0x38, SIG_UINT16(0x0143),       // pushi  [ ignoreActors ]
+	0x78,                           // push1
+	0x76,                           // push0
+	0x81, 0x00,                     // lag 0
+	0x4a, SIG_UINT16(0x000c),       // send c [ GKEgo:setPri: -1, ignoreActors: 0 ]
+	0x33, 0x45,                     // jmp 45 [ end of method ]
+	SIG_MAGICDWORD,
+	0x3c,                           // dup
+	0x35, 0x05,                     // ldi 5
+	0x1a,                           // eq?
+	0x31, 0x3f,                     // bnt 3f [ end of method ]
+	SIG_END
+};
+
+static const uint16 gk1PoliceEgoSpeedFixV1Patch[] = {
+	0x31, 0x1b,                     // bnt 1b [ state 5 ]
+	SIG_ADDTOOFFSET(+19),
+	0x81, 0x00,                     // lag 0
+	0x4a, PATCH_UINT16(0x0006),     // send 6 [ GKEgo:setPri: -1 ]
+	0x3a,                           // toss
+	0x48,                           // ret
+	0x33, 0x00,                     // jmp 0 [ waste 2 bytes ]
+	0x38, PATCH_UINT16(0x0300),     // pushi 0300 [ normalize ]
```

Comment:
Please use PATCH_SELECTOR16() for patches like this.
Otherwise the value is hardcoded and if there is a version that doesn't use the same values, it won't work.